.github/mergify.yml

```diff
@@ -21,24 +21,46 @@ pull_request_rules:
         - check-success=actionlint
       - -files~=.github/.*$
 
+    # Specific condition to allow merging if only specific infra script files are changed
+    - or:
+      - and:
+        - files=scripts/infra/cloud-instance.sh
+        - -check-success=lint
+        - -check-success=test-workflow-complete
+        - check-success=shellcheck
+      - and:
+        - files~=scripts/*.sh$
+        - check-success=shellcheck
+      - and:
+        - -files~=scripts/*.sh$
+        - -files~=.*\.py$
+        - -files~=pyproject.toml$
+        - -files~=requirements.*\.txt$
+        - -files~=tox.ini$
+
     # lint and test must pass if files change that would trigger this job
     - or:
       - and:
         - check-success=lint
         - check-success=test-workflow-complete
+        - -files=scripts/infra/cloud-instance.sh
         - or:
           # see .github/workflows/lint.yml and test.yml
           - files~=.*\.py$
           - files~=pyproject.toml$
           - files~=requirements.*\.txt$
           - files~=tox.ini$
-          - files~=scripts/.*\.sh$
+        # the following line only parses the shell scripts in the scripts directory but not the
+        # infra scripts - it's doing a depth 1 search
+          - files~=scripts/*.sh$
```

Comment:
I don't think this is a good regex.

I think the simple solution is just to just change the lint's regex for .sh to

```
- files~=scripts/[^/]+\.sh$
```

This would properly exclude subfolders of scripts.

It would be nice if mergify config has a way of using globs instead of regex. Then is would be much easier to keep in sync with the workflow's globs.

Reply:
Why isn't it a good regex? `scripts/*.sh$` excludes subfolders, it's a simple shell glob.

Other than I like your proposal, this is a simpler change. Thanks!

Reply:
`scripts/*.sh$` means `scripts` followed by zero-or-more `/` (`/*`) followed by any character (`.`) followed by `sh` followed by end-of-string. So `scriptsxsh` would match as would `scripts/xsh`. I don't think this is the intention.

Shell globs are not regex. The `~=` operated is followed by a regex not a glob.

Reply:
> `scripts/*.sh$` means `scripts` followed by zero-or-more `/` (`/*`) followed by any character (`.`) followed by `sh` followed by end-of-string. So `scriptsxsh` would match as would `scripts/xsh`. `scripts/ruff.sh` would not match. I don't think this is the intention.

That's correct, thanks!

> 
> Shell globs are not regex. The [`~=` operator](https://docs.mergify.com/configuration/conditions/#operators-list) is followed by a regex not a glob.

Thanks for the pointer, I did not realize that!